packages/web_benchmarks/lib/src/browser.dart

```diff
@@ -152,25 +151,26 @@ class Chrome {
     // Subscribe to tracing events prior to calling "Tracing.start". Otherwise,
     // we'll miss tracing data.
     _tracingSubscription =
-        _debugConnection.onNotification.listen((WipEvent event) {
+        _debugConnection?.onNotification.listen((WipEvent event) {
       // We receive data as a sequence of "Tracing.dataCollected" followed by
       // "Tracing.tracingComplete" at the end. Until "Tracing.tracingComplete"
       // is received, the data may be incomplete.
       if (event.method == 'Tracing.tracingComplete') {
-        _tracingCompleter.complete();
-        _tracingSubscription.cancel();
+        _tracingCompleter!.complete();
+        _tracingSubscription!.cancel();
         _tracingSubscription = null;
       } else if (event.method == 'Tracing.dataCollected') {
-        final dynamic value = event.params['value'];
+        final dynamic value = event.params!['value'];
         if (value is! List) {
           throw FormatException(
               '"Tracing.dataCollected" returned malformed data. '
               'Expected a List but got: ${value.runtimeType}');
         }
-        _tracingData.addAll(event.params['value'].cast<Map<String, dynamic>>());
+        _tracingData!
```

Comment:
Why is this safe to force-unwrap?

Reply:
initialized [above](https://github.com/flutter/packages/blob/9ab6bea3d178e0bc8fa4d30acd25bda29e31f9e0/packages/web_benchmarks/lib/src/browser.dart#L147)

Reply:
There's enough separation here and for `_tracingCompleter` that this is fragile; it would be safer to make new non-nullable locals for both, and then set the fields to the local variable.